internal/code/imports.go

```diff
@@ -92,24 +90,39 @@ func ImportPathForDir(dir string) (res string) {
 
 var modregex = regexp.MustCompile("module (.*)\n")
 
+type NameForPackage struct {
+	cache    *sync.Map
```

Comment:
Why did this need to be moved out of a private global?

Reply:
Technically it didn't need to be moved, but it was weird to me to use global state when we were already creating a struct to hold the packages field.

In addition to the fact that global state is a dangerous anti-pattern, I suspect that this would also have caused issues if I wanted to use the same binary to generate multiple graphql servers.